keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -3,111 +3,556 @@
+
+<!-- TODO
+Add info on concerned test in below form as read from https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+- `<package>`: `<date>` - `<test coverage>`
+ -->
+
+##### Integration tests
+
+- Create a cluster with ExternalJWTSigner to configure and verify the health status of core components as published by kube-apiserver.
+
+##### e2e tests
+
+- Create a cluster with ExternalJWTSigner configured.
+- Request a token for a service account principal.
+- Use a token as bearer for making requests to kube-apiserver and ensure it succeeds.
+
+### Graduation Criteria
 
 <!-- TODO -->
 
+#### Alpha
+
+#### Beta
+
+#### GA
+
+### Upgrade/Downgrade Strategy
+
+Existing clusters preserving existing functionality: 
+- Need no changes.
+
+Existing clusters using enhanced functionality: 
+- Need a MasterVM workload that implements ExternalJWTSigner
+- Need to configure the cluster with the `--service-account-signing-endpoint` flag. 
+
+
+### Version Skew Strategy
+
+N/A (The change will completely live in Control plane)
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+Feature is not intended to be a toggleable. However, each distribution can decide for themselves.
+
+For distributions that require capability to enable the feature on a live cluster:
+- ExternalJWTSigner needs to combine the supported keys that are listest from external key management solutions and the keys that were on the cluster before enablement.
+- Control plan will need a re-start with `--service-account-key-file` being un-set and `--service-account-signing-endpoint` being set.
+- former value of `--service-account-key-file` (a path on the file system) will need to be supplied to ExternalJWTSigner in some fashion(this is up to the owners of the distribution).
+
+###### Does enabling the feature change any default behavior?
+
+Not from an end-user perspective. 
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Feature is not intended to be a toggleable. However, each distribution can decide for themselves.
+
+Note: While enabling the feature on a live cluster is easier (owing to custom logic in ExternalJWTSigner's implementation), disabling might not make for a good experience because there will be JWTs signed by external keys that are still hanging around. Disabling will either require the supported external keys to be moved onto the disk and be configured via `--service-account-key-file` or the respective distribution will need to come with a fair warning that all active service account tokens will get invalidated.
+
+###### What happens if we re-enable the feature if it was previously rolled back?
+
+experience would be the same as when enabling.
+
+###### Are there any tests for feature enablement/disablement?
+
+<<[UNRESOLVED]>>
+
+Feature is not intended to be a toggleable
```

Comment:
This is a non-starter.  You always have to be able to go back, and in this case I don't even this it is that hard to go back since you just need the public key for the externally signed tokens to keep verifying them.

Reply:
It is possible to achieve; Reworded with hints on how to...